plugin/app/feishu/feishu.go

```diff
@@ -0,0 +1,426 @@
+	"form": {
+		"form_content": "[{\"id\":\"1\", \"type\": \"input\", \"name\":\"@i18n@widget1\"},{\"id\":\"2\", \"type\": \"input\", \"name\":\"@i18n@widget2\"},{\"id\":\"3\", \"type\": \"textarea\", \"name\":\"@i18n@widget3\"}]"
+	},
+	"i18n_resources": [
+		{
+			"is_default": "true",
+			"locale": "zh-CN",
+			"texts": [
+				{
+					"key": "@i18n@approval_name",
+					"value": "Bytebase 工单"
+				},
+				{
+					"key": "@i18n@node_name",
+					"value": "审批"
+				},
+				{
+					"key": "@i18n@widget1",
+					"value": "Issue"
+				},
+				{
+					"key": "@i18n@widget2",
+					"value": "Stage"
+				},
+				{
+					"key": "@i18n@widget3",
+					"value": "Description"
+				}
+			]
+		}
+	],
+	"node_list": [
+		{
+			"id": "START"
+		},
+		{
+			"id": "approve-here",
+			"name": "@i18n@node_name",
+			"approver": [
+				{
+					"type": "Free"
+				}
+			]
+		},
+		{
+			"id": "END"
+		}
+	],
+	"viewers": [
+		{
+			"viewer_type": "NONE"
+		}
+	]
+}`
+	cancelExternalApprovalReq = `{
+  "approval_code": "%s",
+  "instance_code": "%s",
+  "user_id": "%s"
+}`
+)
+
+// GetTenantAccessToken gets tenant access token.
+func (p *feishuProvider) GetTenantAccessToken(appID string, appSecret string) (*TenantAccessTokenResponse, error) {
+	body := strings.NewReader(fmt.Sprintf(getTenantAccessTokenReq, appID, appSecret))
+	req, err := http.NewRequest("POST", "https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal", body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response TenantAccessTokenResponse
+	if err := json.Unmarshal(b, &response); err != nil {
+		return nil, err
+	}
+	if response.Code != 0 {
+		return nil, errors.Errorf("failed to get tenant access token: %s", response.Msg)
+	}
+
+	return &response, nil
+}
+
+// CreateApprovalDefinition creates an approval definition.
+func (p *feishuProvider) CreateApprovalDefinition(approvalCode string) (*ApprovalDefinitionResponse, error) {
+	body := strings.NewReader(fmt.Sprintf(createApprovalDefinitionReq, approvalCode))
+	req, err := http.NewRequest("POST", "https://open.feishu.cn/open-apis/approval/v4/approvals", body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+p.Token)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response ApprovalDefinitionResponse
+	if err := json.Unmarshal(b, &response); err != nil {
+		return nil, err
+	}
+
+	if response.Code != 0 {
+		return nil, errors.Errorf("failed to create approval definition: %s", response.Msg)
+	}
+
+	return &response, nil
+}
+
+// CreateExternalApproval creates an approval instance.
```

Comment:
Let's make a comment on the method about what creatorID and approverID are what their formats look like. Possibly link to API documentation.